src/plugins/data/common/exports/export_csv.tsx

```diff
@@ -61,14 +82,36 @@ export function datatableToCSV(
   }
 
   // Convert the array of row objects to an array of row arrays
-  const csvRows = rows.map((row) => {
-    return sortedColumnIds.map((id) =>
-      escapeValues(raw ? row[id] : formatters[id].convert(row[id]))
-    );
-  });
+  const csvRows = rows
+    .map((row) => {
+      return sortedColumnIds.map((id) =>
+        escapeValues(raw ? row[id] : formatters[id].convert(row[id]))
+      );
+    })
+    .sort(rowSortPredicate(sortedColumnIds, columnSorting));
 
   return (
     [header, ...csvRows].map((row) => row.join(csvSeparator)).join(LINE_FEED_CHARACTER) +
     LINE_FEED_CHARACTER
   ); // Add \r\n after last line
 }
+
+function rowSortPredicate(
+  sortedColumnIds: string[],
+  columnSorting?: EuiDataGridColumnSortingConfig[]
+) {
+  if (!columnSorting) return () => 0;
+
+  const columnIdMap = new Map(columnSorting.map(({ id }) => [id, sortedColumnIds.indexOf(id)]));
```

Comment:
This is quadratic. Can we make leverage a lookup map?

Reply:
Removed in 44baf770e2d1e31ee54df6c5fd07448839f5b8ac, in favor of `@kbn/sort-predicates` utilities.